Source/Search/Mixins/SearchProviderMixin.lua

```diff
@@ -0,0 +1,70 @@
+AuctionatorSearchProviderMixin = {}
+
+-- Derive
+function AuctionatorSearchProviderMixin:OnSearchEventReceived(eventName, ...)
+end
+
+-- Derive
+function AuctionatorSearchProviderMixin:CreateSearchTerm(term)
+end
+
+-- Derive
+function AuctionatorSearchProviderMixin:GetSearchProvider()
+end
```

Comment:
The undercut search will have 2 search providers, one for commodities (ores, cloth, ...) and one for items (gear, mounts, ...) - new AH stuff :)

Reply:
Hmmm I'll need to think about how to do multi-multisearch 🙄

Reply:
It's not actually that bad. Basically 2 parts, search for your auctions, and using those results get the undercut check with one search for each, so it's one extra step, which should be possible to hide.

Having 2 search provides will be the only awkward bit